src/components/common/GProfilerModal.tsx

```diff
@@ -0,0 +1,335 @@
+    };
+
+    constructor(props: IGProfilerPopupProps & RouteComponentProps) {
+        super(props);
+        this.state = INITIAL_STATE;
+    }
+
+    openModal = () => {
+        this.setState({
+            ...INITIAL_STATE,
+            showModal: true,
+        });
+    };
+
+    closeModal = () => {
+        this.setState({ showModal: false });
+    };
+
+    render() {
+        const {
+            showModal,
+            topNumFeatures,
+            selectedOrganism,
+            gProfilerToken,
+        } = this.state;
+        const { clusteringID, clusterID } = this.props;
+
+        const handleOpenModal = () => {
+            this.openModal();
+        };
+
+        const handleCloseModal = () => {
+            this.closeModal();
+        };
+
+        const topNumFeaturesArray = [
+            100,
+            200,
+            300,
+            400,
+            500,
+        ].filter((topNumFeaturesValue) =>
+            topNumFeaturesValue < this.props.numFeatures ? true : false
+        );
+
+        const handleClickCreateGProfilerLink = async () => {
+            if (
+                (gProfilerToken === null || gProfilerToken === '') &&
+                selectedOrganism === null
+            ) {
+                this.setState({
+                    error: 'Please select an organism',
+                    gProfilerURL: null,
+                });
+                return;
+            }
+            if (topNumFeatures.length == 0) {
+                this.setState({
+                    error:
+                        'No gene list selected. One gene list is at least required. ',
+                    gProfilerURL: null,
+                });
+                return;
+            }
+
+            if (topNumFeatures.length == 0) {
+                this.setState({
+                    error:
+                        'No gene list selected. One gene list is at least required. ',
+                    gProfilerURL: null,
+                });
+                return;
+            }
+
+            const gProfilerLinkResponse = await BackendAPI.getGProfilerLink(
+                clusteringID,
+                clusterID,
+                topNumFeatures,
+                selectedOrganism,
+                gProfilerToken
+            );
+
+            if (gProfilerLinkResponse.url.length > 8000) {
+                this.setState({
+                    error:
+                        'Too many genes in total. Try to select a combination of gene lists with fewer genes.',
+                    gProfilerURL: null,
+                });
+                return;
+            }
+
+            this.setState({
+                gProfilerURL: gProfilerLinkResponse.url,
+            });
+        };
+
+        const organisms = [
+            { key: 1, text: 'Human', value: 'hsapiens' },
+            { key: 2, text: 'Mouse', value: 'mmusculus' },
+            { key: 3, text: 'Drosophila melanogaster', value: 'dmelanogaster' },
+        ];
```

Comment:
gProfiler supports more organisms than this and they have an API for getting that list from them (See: https://biit.cs.ut.ee/gprofiler/page/apis)

It would be nice if we could allow all of these organisms to work.

Regardless, we should use either full species names (Homo Sapeins, Mus Musculus, Drosophila Melanogaster) OR common names (Human, Mouse, Fly) rather than a combination (I prefer Species names).

Reply:
Yeah I know but I think for now we constrain it to these main species no ?

Reply:
Because there's no rush. I'd prefer this be done "properly" the first time.